### PR TITLE
fix(tag)!: update tag color tokens

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description

This PR updated the Tag component to use new color tokens. 
- use new tokens for selected, not emphasized background colors 
- Updated content color for selected not emphasized to be `gray-50` in order to match XD file 
- Change instances of `--spectrum-Tags` to be `--spectrum-Tag` 
- Updated storybook to include variants: with Avatar, with Icon, and Removable 

Breaking Change:
- updated two class names from `--spectrum-Tags-*` to `--spectrum-Tag` 

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

Test outline:
  1. Open the [deploy url](https://pr-2039--spectrum-css.netlify.app/tag.html) or storybook for the tag component:
- [x] Inspect the selected variant (not emphasized) 
- [x] Check selected, not emphasized background colors are using:

  - neutral-background-color-selected-default
  - neutral-background-color-selected-hover
  - neutral-background-color-selected-down
  - neutral-background-color-selected-key-focus

- [x] check label, icon, and cross icon color (selected not emphasized), should be using `--spectrum-gray-50` 

 2. Open the [storybook](https://pr-2039--spectrum-css.netlify.app/preview/?path=/docs/components-tag--docs) for the tag component. Check that each variant is displayed properly and controls work as expected: 

- [x] Default
- [x] Removable  
- [x] Icon 
- [x] Avatar 

### Regression testing

Validate:

1. A legacy documentation page (such as [accordion](https://pr-###--spectrum-css.netlify.app/accordion.html)), including:

- [x] The page renders correctly
- [x] The page is accessible
- [x] The page is responsive

2. A migrated documentation page (such as [action group](https://pr-###--spectrum-css.netlify.app/actiongroup.html)), including:

- [x] The page renders correctly
- [x] The page is accessible
- [x] The page is responsive

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.

- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.

- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/tag/stories/template.js [hunk 1]
```diff
@@ -55,15 +55,15 @@ export const Template = ({
 						size: "50",
 				  })
 				: ""}
-			${iconName
+			${iconName && !avatarUrl
```
Comment (mlogsdon18) [inline components/tag/stories/template.js:58]: If I try to add an icon in Storybook on the Avatar variant, both the avatar and the icon disappear. The Avatar isn't an available control in Storybook so we should probably remove the icon from showing up as a control for the Avatar variation if that's not allowed to be added

File: components/tag/index.css [hunk 2]
```diff
@@ -26,8 +26,12 @@ governing permissions and limitations under the License.
   --spectrum-tag-label-line-height: var(--spectrum-line-height-100);
   --spectrum-tag-label-font-weight: var(--spectrum-regular-font-weight);
 
-  /* selected content */
-  --spectrum-tag-content-color-selected: var(--spectrum-white);
+  /* selected */
+  --spectrum-tag-content-color-selected: var(--spectrum-gray-50);
```
Comment (castastrophe) [inline components/tag/index.css:30]: Should we add a note or TODO comment suggesting that mapping to global variables is a temporary solution until the design tokens for this are available?

File: components/tag/stories/tag.stories.js [hunk 3]
```diff
@@ -91,12 +91,12 @@ export default {
 		rootClass: "spectrum-Tag",
 		size: "m",
 		label: "Tag label",
-		avatarUrl: "example-ava.png",
+		avatarUrl: undefined,
```
Comment (castastrophe) [inline components/tag/stories/tag.stories.js:94]: If it's undefined, you can remove it from this list entirely.

File: components/tag/stories/template.js (hunk 1 above)
Comment (castastrophe) [reply components/tag/stories/template.js:58]: Per @mlogsdon18's feedback, you can add `if: { arg: 'iconName', truthy: false },` to avatarUrl and `if: { arg: 'avatarUrl', truthy: false },` to iconName to make them mutually exclusive

File: components/tag/index.css (hunk 2 above)
Comment (jenndiaz) [reply components/tag/index.css:30]: in this case this component has already been migrated and the XD file listed `--spectrum-gray-50` as the token

**Comment by jenndiaz** [comment]
> I'd like to see test cases in the PR description for the altered properties when you have a chance. I'm happy to run through them when they're ready.

@castastrophe I added test cases and address the storybook issue.

**Comment by jawinn** [comment]
> Change instances of --spectrum-Tags to be --spectrum-Tag

Just wanted to note that this would also require a change with the [SWC config](https://github.com/adobe/spectrum-web-components/blob/8428cadce83f2dd12b14bc29a28d6c43a6432bf0/packages/tags/src/spectrum-config.js#L82C1-L82C1) that converts that class, so at least two lines would need to change there. So would that make this a breaking change when it merges? I was just looking at wanting to change this plural naming while working on TagGroup; thanks for updating.

**Comment by jenndiaz** [comment]
> > Change instances of --spectrum-Tags to be --spectrum-Tag
> 
> Just wanted to note that this would also require a change with the [SWC config](https://github.com/adobe/spectrum-web-components/blob/8428cadce83f2dd12b14bc29a28d6c43a6432bf0/packages/tags/src/spectrum-config.js#L82C1-L82C1) that converts that class, so at least two lines would need to change there. So would that make this a breaking change when it merges? I was just looking at wanting to change this plural naming while working on TagGroup; thanks for updating.


@jawinn If you are writing up a ticket to address the pluralizations separately, I'm happy to remove them and have them all addressed at once

**Review by castastrophe — APPROVED** [review]

**Pull request merged**